addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -0,0 +1,581 @@
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.openhab.binding.loxone.config.LoxoneMiniserverConfig;
+import org.openhab.binding.loxone.core.LxCategory;
+import org.openhab.binding.loxone.core.LxContainer;
+import org.openhab.binding.loxone.core.LxControl;
+import org.openhab.binding.loxone.core.LxControlInfoOnlyAnalog;
+import org.openhab.binding.loxone.core.LxControlInfoOnlyDigital;
+import org.openhab.binding.loxone.core.LxControlJalousie;
+import org.openhab.binding.loxone.core.LxControlLightController;
+import org.openhab.binding.loxone.core.LxControlPushbutton;
+import org.openhab.binding.loxone.core.LxControlRadio;
+import org.openhab.binding.loxone.core.LxControlSwitch;
+import org.openhab.binding.loxone.core.LxControlTextState;
+import org.openhab.binding.loxone.core.LxOfflineReason;
+import org.openhab.binding.loxone.core.LxServer;
+import org.openhab.binding.loxone.core.LxServerListener;
+import org.openhab.binding.loxone.internal.LoxoneHandlerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Representation of a Loxone Miniserver. It is an openHAB {@link Thing}, which is used to communicate with
+ * objects (controls) configured in the Miniserver over {@link Channels}.
+ *
+ * @author Pawel Pieczul - Initial contribution
+ */
+public class LoxoneMiniserverHandler extends BaseThingHandler implements LxServerListener {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_MINISERVER);
+
+    private LxServer server;
+    private LoxoneHandlerFactory factory;
+    private ChannelTypeUID switchTypeId, roSwitchTypeId, rollerTypeId, infoTypeId;
+    private Logger logger = LoggerFactory.getLogger(LoxoneMiniserverHandler.class);
+    private Map<ChannelUID, LxControl> controls = new HashMap<>();
+
+    /**
+     * Create {@link LoxoneMiniserverHandler} object
+     *
+     * @param thing
+     *            Thing object that creates the handler
+     * @param factory
+     *            factory that creates the handler
+     */
+    public LoxoneMiniserverHandler(Thing thing, LoxoneHandlerFactory factory) {
+        super(thing);
+        this.factory = factory;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (server == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No server attached to this thing");
+            return;
+        }
+
+        LxControl control = getControlFromChannelUID(channelUID);
+        if (control == null) {
+            logger.error("Received command {} from unknown control.", command);
```

Comment:
Agree, sounds like an error situation - but yes, it could happen that the framework sends commands for not existing channels: If there is a link defined to a certain channel id, afaik the framework will pass this on; can be discussed whether this is something where a safeguard should be added in the framework...